daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -132,19 +132,19 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
     */
   def reinterpret(
       submitters: Set[Party],
```

Comment:
Canton currently uses GenActionNode.requiredAuthorizers as the submitters which we get as a side effect of building nodes which let us use the logic in `NodeInfo.requiredAuthorizers`. Do you see a way that we can avoid duplicating the `NodeInfo.requiredAuthorizers` mapping?:

- create: submitters := signatories
- exercise and fetch: submitters := actingParties
- lookup by key: submitters := maintainers

Reply:
@oliverse-da I’m not following. `requiredAuthorizers` just picks the value from the node. If you can currently construct the node, why can’t you pass in the value you use when building the node as the `submitters`?

We can’t compute the authorizers from the command without LF interpretation and imho we shouldn’t do that here. The submitters should really be stored separately.

Reply:
we can continue contructing the node - I was only considering the option of canton turning the canton `ActionDescription` directly into `lf.command.Command` rather than first cooking up a `GenActionNode`.